controllers/componentdetectionquery_controller.go

```diff
@@ -157,32 +167,144 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 
 		if source.DevfileURL == "" {
 			log.Info(fmt.Sprintf("Attempting to read a devfile from the URL %s... %v", source.URL, req.NamespacedName))
+			var devfilesMapReturned map[string][]byte
+			var devfilesURLMapReturned, dockerfileContextMapReturned map[string]string
+			var componentPortsMapReturned map[string][]int
+			revision := source.Revision
+
+			if r.RunKubernetesJob && !(componentDetectionQuery.Annotations["runCDQAnalysisLocal"] == "true") {
```

Comment:
I think we should explain the use of the `runCDQAnalysisLocal` annotation somewhere.  Maybe add a comment here and in the CONTRIBUTING.MD to explain it's used as a flag to run tests in a go module?

Reply:
updated in the description.  thinking if the info is too detailed to be in CONTRIBUTING.MD.